lib/game.rb

```diff
@@ -1,60 +1,57 @@
-require_relative "messages"
+require_relative "message"
 require_relative "console"
 require_relative "display"
 require_relative "board"
+require_relative "peg"
+require_relative "computer_player"
+require_relative "human_player"
 
-class Game
 
+class Game
+  HARDCODED_MODE = 1
+  COMPUTER_MODE = 2
+  HUMAN_MODE = 3
   def initialize(display, board)
     @display = display
     @board = board
   end
 
-  def play
+  def play(ship_mode)
     welcome
-    present_board
-    @result1 = turn(board_with_ship)
-    until hit_all?(@result1[0])
-      @result1 = turn(@result1[1])
+    place_ships_on_grid(ship_mode)
+    present_board(@board.grid)
+    until @board.hit_all?(@board.grid)
+      turn(@board.grid)
     end
+    goodbye
   end
 
-  def turn(board)
-    result = []
-    get_coordinates
-    coordinate = board[@row][@column.to_i]
-    outcome = @board.hit_or_miss(board, @row, @column.to_i)
-    new_board = mark_board_with_hit_or_miss(outcome, board)
-    @display.present_board(new_board)
-    all_boats_hit = @board.hit_all(new_board)
-    win_or_retry_message(outcome, all_boats_hit)
-    result << all_boats_hit
-    result << new_board
-    result
+  def place_ships_on_grid(ship_mode)
+    if ship_mode == HARDCODED_MODE
+      @board.grid_with_one_hardcoded_ship
+    elsif ship_mode == COMPUTER_MODE
+      validator = CoordinateValidator.new(@board)
+      computer_player = ComputerPlayer.new(@board, validator)
+      computer_player.generate_ships
+    elsif ship_mode == HUMAN_MODE
+      human_player = HumanPlayer.new(@board, @display)
+      human_player.generate_ships
```

Comment:
what is the difference between human and computer player if they both are behaving like the computer player e.g. ships placed programmatically?

Reply:
@sarahabimay both players have a method that generate ships, but the way they generate ships is different - the computer player places programmatically whilst the human player uses the display to place the ships